### PR TITLE
Admin Panel

### PR DESCRIPTION
**Pull request by shootdot**
Except Drop down menu everything is done. 😀

### CODE REVIEW

**Comment by shootdot** [comment]
@Sank6 For Dropdown I need to edit each routes 😩

**Comment by wqtalha** [comment]
Hey non-admin users can access this panel

How can we do it for admin only

**Comment by shootdot** [comment]
Bot verifiers and Admins users only can access the panel

**Comment by shootdot** [comment]
image
@wqtalha Works perfectly for me.

**Comment by wqtalha** [comment]
okey ty
It would be nice if it appears in the admin panel on approved or rejected bots

**Comment by wqtalha** [comment]
And can you tell what the color of that error text is in the css file?

**Comment by ElemelkyaDima** [comment]
image

**Comment by shootdot** [comment]
@ElemelkyaDima Fill your config.json file. Your role ids are missing on config.json

**Comment by shootdot** [comment]
> And can you tell what the color of that error text is in the css file?
> It would be nice if it appears in the admin panel on approved or rejected bots
What do you mean by these ?

**Comment by Sank6** [comment]
The colours aren't in the css files. They should be in the theme files @wqtalha .
Also @shootdot . Looks nice. I'll have a look through the code first and then pull onto development.

**Comment by shootdot** [comment]
Yea sure

**Comment by ElemelkyaDima** [comment]
image
@shootdot

**Comment by shootdot** [comment]
I  will have a look onto it right now

**Comment by ElemelkyaDima** [comment]
> Я посмотрю на это прямо сейчас

thx

**Comment by shootdot** [comment]
Is the user in the server ?

**Comment by ElemelkyaDima** [comment]
> Есть ли пользователь на сервере Discord?

yes

**Comment by shootdot** [comment]
If the site isn't loading when you are in `/admin`  page then you are not in the discord server.

**Comment by ElemelkyaDima** [comment]
image

**Comment by ElemelkyaDima** [comment]
INTENTS

**Comment by Sank6** [comment]
I don't think its intents. However, it might be better to fetch the user instead of getting from cache.

**Comment by shootdot** [comment]
Yeah I will change to fetch now

**Comment by shootdot** [comment]
I was evaluating and trying to find what was the error

**Comment by Sank6** [comment]
```js
    const member = await req.app.get('client').guilds.cache.get(id).members.fetch(req.user.id);
    if (!admin_user_ids.includes(req.user.id) && !member.roles.cache.has(bot_verifier)) return res.render("403", { user: req.user });
```

**Comment by wqtalha** [comment]
> It would be nice if it appears in the admin panel on approved or rejected bots

I want rejected or approved bots to appear in admin panel

**Comment by shootdot** [comment]
I need to update in api side also

**Comment by shootdot** [comment]
> > It would be nice if it appears in the admin panel on approved or rejected bots
> 
> I want rejected or approved bots to appear in admin panel

That will be abuse to mongodb

**Comment by shootdot** [comment]
Sank6 btw for dropdown how do I do ? I need to update each route for it

**Comment by wqtalha** [comment]
It would be nice if when we approve a bot, it appears in the admin panel that the bot was approved and rejected when we reject it.

I'm writing from the translation, sorry if I'm writing wrong

**Comment by ElemelkyaDima** [comment]
Can you mark bots as 18+ in the admin area?

**Comment by shootdot** [comment]
> Can you mark bots as 18+ in the admin area?

Nope you can do that in discord `[p]marknsfw @bot`

**Comment by ElemelkyaDima** [comment]
> > Можете ли вы пометить ботов как 18+ в админке?
> 
> Нет, ты можешь сделать это в разладе `[p]marknsfw @bot`

ok

**Comment by Sank6** [comment]
You need to import the server id from the config in the `api/admin` file.

**Comment by shootdot** [comment]
Done Thanks

**Comment by Zaid-maker** [comment]
Amazing work man that's freaking awesome that someone just created admin panel for the list. Big ThumbsUp for you. I am also leasrning react-admin-panel and it's really fun.

**Comment by shootdot** [comment]
@Sank6 Changes are done but can u give any idea for dropdown ?

**Comment by Sank6** [comment]
I was thinking editing the `req.user` to add a property staff and then use it across the routes. Dw though, I can do it after merging it into dev.

**Comment by shootdot** [comment]
Yeah

**Comment by shootdot** [comment]
That's a good Idea.

File: src/routes/api/admin/index.js
Comment (Zaid-maker) [inline src/routes/api/admin/index.js:45]: You have forget 'let' before e = new MessageEmbed()

**Review by Zaid-maker — CHANGES_REQUESTED** [review]

File: src/routes/api/admin/index.js
Comment (Sank6) [reply src/routes/api/admin/index.js:45]: I'm going to rewrite that file in multiple files soon with different endpoints for different actions.

Comment (Zaid-maker) [reply src/routes/api/admin/index.js:45]: ok

Comment (shootdot) [reply src/routes/api/admin/index.js:45]: Huh My Bad 😞

Comment (shootdot) [reply src/routes/api/admin/index.js:45]: > I'm going to rewrite that file in multiple files soon with different endpoints for different actions.

Why ? Is the current format bad ?

Comment (Sank6) [reply src/routes/api/admin/index.js:45]: It's just neater to have different endpoints for the different methods.

Comment (shootdot) [reply src/routes/api/admin/index.js:45]: Yea true...

**Comment by FORWENTV** [comment]
Hello I have an error with the admin page in http: // localhost / admin it gives me this error!



TypeError: C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ src \ dynamic \ includes \ nav.pug: 14
    12 | a (href = `/ api / embed / $ {bot.botid}` target = "_ blank")

    13 | span.navtext Embed

  > 14 | if ((req.baseUrl == "/ user" || req.baseUrl == "/ me") && userProfile)

    15 | a (href = `/ user / $ {userProfile.id}`)

    16 | span.navtext # {userProfile.username}

    17 | if (req.baseUrl == "/ tag")


Cannot read property 'baseUrl' of undefined
    at eval (eval at wrap (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ pug-runtime \ wrap.js: 6: 10), <anonymous>: 113: 11)
    at template (eval at wrap (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ pug-runtime \ wrap.js: 6: 10), <anonymous>: 364: 7)
    at Object.exports.renderFile (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ pug \ lib \ index.js: 452: 38)
    at Object.exports.renderFile (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ pug \ lib \ index.js: 442: 21)
    at View.exports .__ express [as engine] (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ pug \ lib \ index.js: 491: 11)
    at View.render (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ express \ lib \ view.js: 135: 8)
    at tryRender (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ express \ lib \ application.js: 640: 10)
    at Function.render (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ express \ lib \ application.js: 592: 3)
    at ServerResponse.render (C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ node_modules \ express \ lib \ response.js: 1012: 7)
    at C: \ Users \ forwen \ Desktop \ bot-list (center) \ Dev \ src \ routes \ admin.js: 16: 9
    at processTicksAndRejections (internal / process / task_queues.js: 97: 5)

Can you help me thank you

**Pull request merged**
